vowpalwabbit/interactions.cc

```diff
@@ -80,6 +80,61 @@ std::vector<std::vector<namespace_index>> expand_interactions(
   return res;
 }
 
+void expand_quadratics_wildcard_interactions(namsepace_interactions& interactions)
+{
+  if (interactions.size == interactions.all_seen_namespaces.size())
+  {
+    // nothing new here
+    return;
+  }
+  interactions.size = interactions.all_seen_namespaces.size();
+  auto set_interactions = interactions.all_seen_namespaces;
+  std::vector<std::vector<namespace_index>> active_interactions;
+  for (auto it = set_interactions.begin(); it != set_interactions.end(); ++it)
+  {
+    for (auto jt = it; jt != set_interactions.end(); ++jt)
+    {
+      if (interactions.active_interactions.find({*it, *jt}) == interactions.active_interactions.end())
+      {
+        active_interactions.push_back({*it, *jt});
+        interactions.active_interactions.insert({*it, *jt});
+      }
+      if (interactions.active_interactions.find({*it, *it}) == interactions.active_interactions.end())
```

Comment:
seems jt-independent - can we extract it from jt loop?

Reply:
Yes good point